The job **`dist-various-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/26799634538/job/79003427462) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/79003427462) [(plain)](https://github.com/rust-lang/rust/commit/d67550959982b2d58ed8b6ab89af2021bdc52127/checks/79003427462/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
DirectMap4k:      150656 kB
DirectMap2M:     7184384 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing python3 ../x.py --stage 2 test --host= --target thumbv6m-none-eabi,thumbv7m-none-eabi,thumbv7em-none-eabi,thumbv7em-none-eabihf tests/run-make tests/run-make-cargo &&       python3 ../x.py dist --host= --target wasm32-unknown-emscripten,arm-unknown-linux-musleabi,arm-unknown-linux-musleabihf,armv5te-unknown-linux-gnueabi,armv5te-unknown-linux-musleabi,armv7-unknown-linux-musleabihf,aarch64-unknown-none,aarch64-unknown-none-softfloat,sparc64-unknown-linux-gnu,x86_64-unknown-redox,thumbv6m-none-eabi,thumbv7m-none-eabi,thumbv7em-none-eabi,thumbv7em-none-eabihf,thumbv8m.base-none-eabi,thumbv8m.main-none-eabi,thumbv8m.main-none-eabihf,riscv32i-unknown-none-elf,riscv32im-unknown-none-elf,riscv32imc-unknown-none-elf,riscv32imac-unknown-none-elf,riscv32imafc-unknown-none-elf,riscv64imac-unknown-none-elf,riscv64gc-unknown-none-elf,armv7r-none-eabi,thumbv7r-none-eabi,armv7r-none-eabihf,thumbv7r-none-eabihf,armv8r-none-eabihf,thumbv8r-none-eabihf,thumbv7neon-unknown-linux-gnueabihf,armv7a-none-eabi,thumbv7a-none-eabi,armv7a-none-eabihf,thumbv7a-none-eabihf
+ python3 ../x.py --stage 2 test --host= --target thumbv6m-none-eabi,thumbv7m-none-eabi,thumbv7em-none-eabi,thumbv7em-none-eabihf tests/run-make tests/run-make-cargo
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.06s
##[endgroup]
downloading https://static.rust-lang.org/dist/2026-05-31/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
##[endgroup]
[TIMING:end] test::Compiletest { test_compiler: Compiler { stage: 2, host: x86_64-unknown-linux-gnu, forced_compiler: false }, target: thumbv7em-none-eabihf, mode: run-make, suite: "run-make-cargo", path: "tests/run-make-cargo", compare_mode: None } -- 54.997
[TIMING:end] test::RunMakeCargo { test_compiler: Compiler { stage: 2, host: x86_64-unknown-linux-gnu, forced_compiler: false }, target: thumbv7em-none-eabihf } -- 0.000
Build completed successfully in 0:56:46
+ python3 ../x.py dist --host= --target wasm32-unknown-emscripten,arm-unknown-linux-musleabi,arm-unknown-linux-musleabihf,armv5te-unknown-linux-gnueabi,armv5te-unknown-linux-musleabi,armv7-unknown-linux-musleabihf,aarch64-unknown-none,aarch64-unknown-none-softfloat,sparc64-unknown-linux-gnu,x86_64-unknown-redox,thumbv6m-none-eabi,thumbv7m-none-eabi,thumbv7em-none-eabi,thumbv7em-none-eabihf,thumbv8m.base-none-eabi,thumbv8m.main-none-eabi,thumbv8m.main-none-eabihf,riscv32i-unknown-none-elf,riscv32im-unknown-none-elf,riscv32imc-unknown-none-elf,riscv32imac-unknown-none-elf,riscv32imafc-unknown-none-elf,riscv64imac-unknown-none-elf,riscv64gc-unknown-none-elf,armv7r-none-eabi,thumbv7r-none-eabi,armv7r-none-eabihf,thumbv7r-none-eabihf,armv8r-none-eabihf,thumbv8r-none-eabihf,thumbv7neon-unknown-linux-gnueabihf,armv7a-none-eabi,thumbv7a-none-eabi,armv7a-none-eabihf,thumbv7a-none-eabihf
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.11s
##[endgroup]
cargo:warning=Compiler family detection failed due to error: ToolNotFound: failed to find tool "sparc64-linux-gnu-g++": No such file or directory (os error 2)
cargo:warning=Compiler family detection failed due to error: ToolNotFound: failed to find tool "sparc64-linux-gnu-g++": No such file or directory (os error 2)
---
[RUSTC-TIMING] cc test:false 0.912
   Compiling core v0.0.0 (/checkout/library/core)
[RUSTC-TIMING] build_script_build test:false 0.287
   Compiling compiler_builtins v0.1.160 (/checkout/library/compiler-builtins/compiler-builtins)
warning: compiler_builtins@0.1.160: /checkout/src/llvm-project/compiler-rt/lib/builtins/arm/fixdfsivfp.S: Assembler messages:
warning: compiler_builtins@0.1.160: /checkout/src/llvm-project/compiler-rt/lib/builtins/arm/fixdfsivfp.S:25: Error: selected FPU does not support instruction -- `vmov d7,r0,r1'
warning: compiler_builtins@0.1.160: /checkout/src/llvm-project/compiler-rt/lib/builtins/arm/fixdfsivfp.S:26: Error: selected processor does not support `vcvt.s32.f64 s15,d7' in Thumb mode
warning: compiler_builtins@0.1.160: /checkout/src/llvm-project/compiler-rt/lib/builtins/arm/fixdfsivfp.S:27: Error: selected FPU does not support instruction -- `vmov r0,s15'
error: failed to run custom build command for `compiler_builtins v0.1.160 (/checkout/library/compiler-builtins/compiler-builtins)`

Caused by:
  process didn't exit successfully: `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-std/dist/build/compiler_builtins-7275cb9ad4c72c0d/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=LIBM_BUILD_VERBOSE
  cargo::rerun-if-changed=build.rs
  cargo::rerun-if-changed=../libm/configure.rs
  cargo:rustc-check-cfg=cfg(thumb)
  cargo:rustc-cfg=thumb
  cargo:rustc-check-cfg=cfg(thumb_1)
  cargo:rustc-check-cfg=cfg(x86_no_sse2)
  cargo:rustc-check-cfg=cfg(assert_no_panic)
  cargo:rustc-check-cfg=cfg(intrinsics_enabled)
  cargo:rustc-cfg=intrinsics_enabled
  cargo:rustc-check-cfg=cfg(optimizations_enabled)
  cargo:rustc-cfg=optimizations_enabled
  cargo:rustc-check-cfg=cfg(f16_enabled)
  cargo:rustc-cfg=f16_enabled
  cargo:rustc-check-cfg=cfg(f128_enabled)
  cargo:rustc-cfg=f128_enabled
  cargo::rustc-check-cfg=cfg(__ashldi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__ashlsi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__ashrdi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__ashrsi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__bswapsi2, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__bswapdi2, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__bswapti2, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__divdi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__divsi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__divmoddi4, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__divmodsi4, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__divmodsi4, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__divmodti4, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__lshrdi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__lshrsi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__moddi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__modsi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__muldi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__udivdi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__udivmoddi4, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__udivmodsi4, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__udivsi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__umoddi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__umodsi3, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas1_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas1_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas1_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas1_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas2_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas2_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas2_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas2_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas4_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas4_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas4_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas4_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas8_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas8_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas8_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas8_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas16_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas16_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas16_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_cas16_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd1_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd1_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd1_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd1_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd2_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd2_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd2_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd2_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd4_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd4_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd4_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd4_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd8_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd8_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd8_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldadd8_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr1_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr1_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr1_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr1_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr2_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr2_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr2_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr2_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr4_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr4_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr4_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr4_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr8_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr8_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr8_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldclr8_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor1_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor1_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor1_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor1_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor2_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor2_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor2_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor2_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor4_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor4_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor4_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor4_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor8_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor8_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor8_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldeor8_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset1_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset1_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset1_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset1_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset2_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset2_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset2_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset2_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset4_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset4_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset4_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset4_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset8_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset8_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset8_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_ldset8_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp1_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp1_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp1_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp1_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp2_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp2_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp2_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp2_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp4_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp4_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp4_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp4_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp8_relax, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp8_acq, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp8_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(__aarch64_swp8_acq_rel, values("optimized-c"))
  cargo::rustc-check-cfg=cfg(target_feature, values("vis3"))
  cargo:compiler-rt=/checkout/library/compiler-builtins/compiler-builtins/compiler-rt
  cargo:rustc-cfg=feature="mem"
  cargo:rustc-check-cfg=cfg(mem_unaligned)
  cargo:rustc-check-cfg=cfg(kernel_user_helpers)
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/absvdi2.c
  cargo:rustc-cfg=__absvdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/absvsi2.c
  cargo:rustc-cfg=__absvsi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/absvti2.c
  cargo:rustc-cfg=__absvti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/addvdi3.c
  cargo:rustc-cfg=__addvdi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/addvsi3.c
  cargo:rustc-cfg=__addvsi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/addvti3.c
  cargo:rustc-cfg=__addvti3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/aeabi_cdcmp.S
  cargo:rustc-cfg=__aeabi_cdcmp="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/aeabi_cdcmpeq_check_nan.c
  cargo:rustc-cfg=__aeabi_cdcmpeq_check_nan="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/aeabi_cfcmp.S
  cargo:rustc-cfg=__aeabi_cfcmp="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/aeabi_cfcmpeq_check_nan.c
  cargo:rustc-cfg=__aeabi_cfcmpeq_check_nan="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/aeabi_div0.c
  cargo:rustc-cfg=__aeabi_div0="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/aeabi_drsub.c
  cargo:rustc-cfg=__aeabi_drsub="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/aeabi_frsub.c
  cargo:rustc-cfg=__aeabi_frsub="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/bswapdi2.S
  cargo:rustc-cfg=__bswapdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/bswapsi2.S
  cargo:rustc-cfg=__bswapsi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/cmpdi2.c
  cargo:rustc-cfg=__cmpdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/cmpti2.c
  cargo:rustc-cfg=__cmpti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/divdc3.c
  cargo:rustc-cfg=__divdc3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/divmodsi4.S
  cargo:rustc-cfg=__divmodsi4="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/divsc3.c
  cargo:rustc-cfg=__divsc3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/divsi3.S
  cargo:rustc-cfg=__divsi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/ffsti2.c
  cargo:rustc-cfg=__ffsti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/fixdfsivfp.S
  cargo:rustc-cfg=__fixdfsivfp="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/fixsfsivfp.S
  cargo:rustc-cfg=__fixsfsivfp="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/fixunsdfsivfp.S
  cargo:rustc-cfg=__fixunsdfsivfp="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/fixunssfsivfp.S
  cargo:rustc-cfg=__fixunssfsivfp="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/floatsidfvfp.S
  cargo:rustc-cfg=__floatsidfvfp="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/floatsisfvfp.S
  cargo:rustc-cfg=__floatsisfvfp="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/floatunssidfvfp.S
  cargo:rustc-cfg=__floatunssidfvfp="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/floatunssisfvfp.S
  cargo:rustc-cfg=__floatunssisfvfp="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/int_util.c
  cargo:rustc-cfg=__int_util="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/modsi3.S
  cargo:rustc-cfg=__modsi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/muldc3.c
  cargo:rustc-cfg=__muldc3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/mulsc3.c
  cargo:rustc-cfg=__mulsc3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/mulvdi3.c
  cargo:rustc-cfg=__mulvdi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/mulvsi3.c
  cargo:rustc-cfg=__mulvsi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/mulvti3.c
  cargo:rustc-cfg=__mulvti3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negdf2.c
  cargo:rustc-cfg=__negdf2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/negdf2vfp.S
  cargo:rustc-cfg=__negdf2vfp="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negdi2.c
  cargo:rustc-cfg=__negdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negsf2.c
  cargo:rustc-cfg=__negsf2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/negsf2vfp.S
  cargo:rustc-cfg=__negsf2vfp="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negti2.c
  cargo:rustc-cfg=__negti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negvdi2.c
  cargo:rustc-cfg=__negvdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negvsi2.c
  cargo:rustc-cfg=__negvsi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/negvti2.c
  cargo:rustc-cfg=__negvti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/paritydi2.c
  cargo:rustc-cfg=__paritydi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/paritysi2.c
  cargo:rustc-cfg=__paritysi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/parityti2.c
  cargo:rustc-cfg=__parityti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/popcountdi2.c
  cargo:rustc-cfg=__popcountdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/popcountsi2.c
  cargo:rustc-cfg=__popcountsi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/popcountti2.c
  cargo:rustc-cfg=__popcountti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/restore_vfp_d8_d15_regs.S
  cargo:rustc-cfg=__restore_vfp_d8_d15_regs="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/save_vfp_d8_d15_regs.S
  cargo:rustc-cfg=__save_vfp_d8_d15_regs="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/subvdi3.c
  cargo:rustc-cfg=__subvdi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/subvsi3.c
  cargo:rustc-cfg=__subvsi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/subvti3.c
  cargo:rustc-cfg=__subvti3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/switch16.S
  cargo:rustc-cfg=__switch16="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/switch32.S
  cargo:rustc-cfg=__switch32="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/switch8.S
  cargo:rustc-cfg=__switch8="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/switchu8.S
  cargo:rustc-cfg=__switchu8="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/sync_synchronize.S
  cargo:rustc-cfg=__sync_synchronize="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/ucmpdi2.c
  cargo:rustc-cfg=__ucmpdi2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/ucmpti2.c
  cargo:rustc-cfg=__ucmpti2="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/udivmodsi4.S
  cargo:rustc-cfg=__udivmodsi4="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/udivsi3.S
  cargo:rustc-cfg=__udivsi3="optimized-c"
  cargo:rerun-if-changed=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/umodsi3.S
  cargo:rustc-cfg=__umodsi3="optimized-c"
  OUT_DIR = Some(/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-std/thumbv7r-none-eabihf/dist/build/compiler_builtins-7c5535102dd3bbfd/out)
  OPT_LEVEL = Some(3)
  TARGET = Some(thumbv7r-none-eabihf)
  HOST = Some(x86_64-unknown-linux-gnu)
  cargo:rerun-if-env-changed=CC_thumbv7r-none-eabihf
  CC_thumbv7r-none-eabihf = None
  cargo:rerun-if-env-changed=CC_thumbv7r_none_eabihf
  CC_thumbv7r_none_eabihf = Some(sccache arm-none-eabi-gcc)
  cargo:rerun-if-env-changed=CC_KNOWN_WRAPPER_CUSTOM
  CC_KNOWN_WRAPPER_CUSTOM = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some(true)
  CARGO_CFG_TARGET_FEATURE = Some(dsp,fpregs,rclass,thumb-mode,thumb2,v5te,v6,v6k,v6t2,v7,vfp2)
  cargo:rerun-if-env-changed=CFLAGS_thumbv7r-none-eabihf
  CFLAGS_thumbv7r-none-eabihf = None
  cargo:rerun-if-env-changed=CFLAGS_thumbv7r_none_eabihf
  CFLAGS_thumbv7r_none_eabihf = Some( -march=armv7-r+fp.sp)
  cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
  CC_SHELL_ESCAPED_FLAGS = None
  OUT_DIR = Some(/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-std/thumbv7r-none-eabihf/dist/build/compiler_builtins-7c5535102dd3bbfd/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  TARGET = Some(thumbv7r-none-eabihf)
  CARGO_CFG_TARGET_FEATURE = Some(dsp,fpregs,rclass,thumb-mode,thumb2,v5te,v6,v6k,v6t2,v7,vfp2)
  HOST = Some(x86_64-unknown-linux-gnu)
  cargo:rerun-if-env-changed=CFLAGS_thumbv7r-none-eabihf
  CFLAGS_thumbv7r-none-eabihf = None
  cargo:rerun-if-env-changed=CFLAGS_thumbv7r_none_eabihf
  CFLAGS_thumbv7r_none_eabihf = Some( -march=armv7-r+fp.sp)
  cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
  CC_SHELL_ESCAPED_FLAGS = None
  OUT_DIR = Some(/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-std/thumbv7r-none-eabihf/dist/build/compiler_builtins-7c5535102dd3bbfd/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  TARGET = Some(thumbv7r-none-eabihf)
  CARGO_CFG_TARGET_FEATURE = Some(dsp,fpregs,rclass,thumb-mode,thumb2,v5te,v6,v6k,v6t2,v7,vfp2)
  HOST = Some(x86_64-unknown-linux-gnu)
  cargo:rerun-if-env-changed=CFLAGS_thumbv7r-none-eabihf
  CFLAGS_thumbv7r-none-eabihf = None
  cargo:rerun-if-env-changed=CFLAGS_thumbv7r_none_eabihf
  CFLAGS_thumbv7r_none_eabihf = Some( -march=armv7-r+fp.sp)
  cargo:rerun-if-env-changed=CC_SHELL_ESCAPED_FLAGS
  CC_SHELL_ESCAPED_FLAGS = None
  cargo:warning=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/fixdfsivfp.S: Assembler messages:
  cargo:warning=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/fixdfsivfp.S:25: Error: selected FPU does not support instruction -- `vmov d7,r0,r1'
  cargo:warning=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/fixdfsivfp.S:26: Error: selected processor does not support `vcvt.s32.f64 s15,d7' in Thumb mode
  cargo:warning=/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/fixdfsivfp.S:27: Error: selected FPU does not support instruction -- `vmov r0,s15'

  --- stderr


  error occurred: Command "sccache" "arm-none-eabi-gcc" "-O3" "-ffunction-sections" "-fdata-sections" "-g" "-fno-omit-frame-pointer" "-mfloat-abi=hard" "-mthumb" "-march=armv7-r+fp.sp" "-fno-builtin" "-fvisibility=hidden" "-ffreestanding" "-Werror=implicit-function-declaration" "-fomit-frame-pointer" "-ffile-prefix-map=/checkout/src/llvm-project/compiler-rt=." "-DCOMPILER_RT_HAS_FLOAT16" "-DVISIBILITY_HIDDEN" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-std/thumbv7r-none-eabihf/dist/build/compiler_builtins-7c5535102dd3bbfd/out/63f8a0b8a638da8c-fixdfsivfp.o" "-c" "/checkout/src/llvm-project/compiler-rt/lib/builtins/arm/fixdfsivfp.S" with args arm-none-eabi-gcc did not execute successfully (status code exit status: 1).


warning: build failed, waiting for other jobs to finish...
[RUSTC-TIMING] core test:false 19.752
Bootstrap failed while executing `dist --host= --target wasm32-unknown-emscripten,arm-unknown-linux-musleabi,arm-unknown-linux-musleabihf,armv5te-unknown-linux-gnueabi,armv5te-unknown-linux-musleabi,armv7-unknown-linux-musleabihf,aarch64-unknown-none,aarch64-unknown-none-softfloat,sparc64-unknown-linux-gnu,x86_64-unknown-redox,thumbv6m-none-eabi,thumbv7m-none-eabi,thumbv7em-none-eabi,thumbv7em-none-eabihf,thumbv8m.base-none-eabi,thumbv8m.main-none-eabi,thumbv8m.main-none-eabihf,riscv32i-unknown-none-elf,riscv32im-unknown-none-elf,riscv32imc-unknown-none-elf,riscv32imac-unknown-none-elf,riscv32imafc-unknown-none-elf,riscv64imac-unknown-none-elf,riscv64gc-unknown-none-elf,armv7r-none-eabi,thumbv7r-none-eabi,armv7r-none-eabihf,thumbv7r-none-eabihf,armv8r-none-eabihf,thumbv8r-none-eabihf,thumbv7neon-unknown-linux-gnueabihf,armv7a-none-eabi,thumbv7a-none-eabi,armv7a-none-eabihf,thumbv7a-none-eabihf`
Build completed unsuccessfully in 0:18:25
  local time: Tue Jun  2 06:34:11 UTC 2026
  network time: Tue, 02 Jun 2026 06:34:11 GMT
##[error]Process completed with exit code 1.
##[group]Run echo "disk usage:"
```

</details>

--- a/compiler/rustc_target/src/spec/targets/thumbv7a_none_eabi.rs
+++ b/compiler/rustc_target/src/spec/targets/thumbv7a_none_eabi.rs
@@ -7,7 +7,7 @@ pub(crate) fn target() -> Target {
         llvm_target: "thumbv7a-none-eabi".into(),
         metadata: TargetMetadata {
             description: Some("Thumb-mode Bare Armv7-A".into()),
-            tier: Some(3),
+            tier: Some(2),
             host_tools: Some(false),
             std: Some(false),
         },

--- a/compiler/rustc_target/src/spec/targets/thumbv7a_none_eabihf.rs
+++ b/compiler/rustc_target/src/spec/targets/thumbv7a_none_eabihf.rs
@@ -7,7 +7,7 @@ pub(crate) fn target() -> Target {
         llvm_target: "thumbv7a-none-eabihf".into(),
         metadata: TargetMetadata {
             description: Some("Thumb-mode Bare Armv7-A, hardfloat".into()),
-            tier: Some(3),
+            tier: Some(2),
             host_tools: Some(false),
             std: Some(false),
         },

--- a/compiler/rustc_target/src/spec/targets/thumbv7r_none_eabi.rs
+++ b/compiler/rustc_target/src/spec/targets/thumbv7r_none_eabi.rs
@@ -7,7 +7,7 @@ pub(crate) fn target() -> Target {
         llvm_target: "thumbv7r-none-eabi".into(),
         metadata: TargetMetadata {
             description: Some("Thumb-mode Bare Armv7-R".into()),
-            tier: Some(3),
+            tier: Some(2),
             host_tools: Some(false),
             std: Some(false),
         },

--- a/compiler/rustc_target/src/spec/targets/thumbv7r_none_eabihf.rs
+++ b/compiler/rustc_target/src/spec/targets/thumbv7r_none_eabihf.rs
@@ -7,7 +7,7 @@ pub(crate) fn target() -> Target {
         llvm_target: "thumbv7r-none-eabihf".into(),
         metadata: TargetMetadata {
             description: Some("Thumb-mode Bare Armv7-R, hardfloat".into()),
-            tier: Some(3),
+            tier: Some(2),
             host_tools: Some(false),
             std: Some(false),
         },

--- a/compiler/rustc_target/src/spec/targets/thumbv8r_none_eabihf.rs
+++ b/compiler/rustc_target/src/spec/targets/thumbv8r_none_eabihf.rs
@@ -7,7 +7,7 @@ pub(crate) fn target() -> Target {
         llvm_target: "thumbv8r-none-eabihf".into(),
         metadata: TargetMetadata {
             description: Some("Thumb-mode Bare Armv8-R, hardfloat".into()),
-            tier: Some(3),
+            tier: Some(2),
             host_tools: Some(false),
             std: Some(false),
         },

--- a/src/ci/docker/host-x86_64/dist-various-1/Dockerfile
+++ b/src/ci/docker/host-x86_64/dist-various-1/Dockerfile
@@ -114,11 +114,16 @@ ENV TARGETS=$TARGETS,riscv32imafc-unknown-none-elf
 ENV TARGETS=$TARGETS,riscv64imac-unknown-none-elf
 ENV TARGETS=$TARGETS,riscv64gc-unknown-none-elf
 ENV TARGETS=$TARGETS,armv7r-none-eabi
+ENV TARGETS=$TARGETS,thumbv7r-none-eabi
 ENV TARGETS=$TARGETS,armv7r-none-eabihf
+ENV TARGETS=$TARGETS,thumbv7r-none-eabihf
 ENV TARGETS=$TARGETS,armv8r-none-eabihf
+ENV TARGETS=$TARGETS,thumbv8r-none-eabihf
 ENV TARGETS=$TARGETS,thumbv7neon-unknown-linux-gnueabihf
 ENV TARGETS=$TARGETS,armv7a-none-eabi
+ENV TARGETS=$TARGETS,thumbv7a-none-eabi
 ENV TARGETS=$TARGETS,armv7a-none-eabihf
+ENV TARGETS=$TARGETS,thumbv7a-none-eabihf
 
 ENV CFLAGS_armv5te_unknown_linux_musleabi="-march=armv5te -marm -mfloat-abi=soft" \
     CFLAGS_arm_unknown_linux_musleabi="-march=armv6 -marm" \
@@ -133,8 +138,18 @@ ENV CFLAGS_armv5te_unknown_linux_musleabi="-march=armv5te -marm -mfloat-abi=soft
     CC_armv7a_none_eabihf=arm-none-eabi-gcc \
     CFLAGS_armv7a_none_eabi=-march=armv7-a \
     CFLAGS_armv7a_none_eabihf=-march=armv7-a+fp \
+    CC_thumbv7a_none_eabi=arm-none-eabi-gcc \
+    CC_thumbv7a_none_eabihf=arm-none-eabi-gcc \
+    CFLAGS_thumbv7a_none_eabi=-march=armv7-a \
+    CFLAGS_thumbv7a_none_eabihf="-march=armv7-a+fp -mfpu=vfpv3-d16" \
+    CC_thumbv7r_none_eabi=arm-none-eabi-gcc \
+    CC_thumbv7r_none_eabihf=arm-none-eabi-gcc \
+    CFLAGS_thumbv7r_none_eabi=-march=armv7-r \
+    CFLAGS_thumbv7r_none_eabihf="-march=armv7-r+fp -mfpu=vfpv3-d16" \
     CC_armv8r_none_eabihf=arm-none-eabi-gcc \
     CFLAGS_armv8r_none_eabihf="-march=armv8-r+fp.sp -mfpu=fp-armv8" \
+    CC_thumbv8r_none_eabihf=arm-none-eabi-gcc \
+    CFLAGS_thumbv8r_none_eabihf="-march=armv8-r+fp.sp -mfpu=fp-armv8" \
     CC_aarch64_unknown_none_softfloat=aarch64-none-elf-gcc \
     CFLAGS_aarch64_unknown_none_softfloat=-mstrict-align -march=armv8-a+nofp+nosimd \
     CC_aarch64_unknown_none=aarch64-none-elf-gcc \

--- a/src/doc/rustc/src/platform-support.md
+++ b/src/doc/rustc/src/platform-support.md
@@ -174,6 +174,11 @@ target | std | notes
 [`armv7r-none-eabi`](platform-support/armv7r-none-eabi.md) | * | Bare Armv7-R
 [`armv7r-none-eabihf`](platform-support/armv7r-none-eabi.md) | * | Bare Armv7-R, hardfloat
 [`armv8r-none-eabihf`](platform-support/armv8r-none-eabihf.md) | * | Bare Armv8-R, hardfloat
+[`thumbv7a-none-eabi`](platform-support/armv7a-none-eabi.md) | * | Thumb-mode Bare Armv7-A
+[`thumbv7a-none-eabihf`](platform-support/armv7a-none-eabi.md) | * | Thumb-mode Bare Armv7-A, hardfloat
+[`thumbv7r-none-eabi`](platform-support/armv7r-none-eabi.md) | * | Thumb-mode Bare Armv7-R
+[`thumbv7r-none-eabihf`](platform-support/armv7r-none-eabi.md) | * | Thumb-mode Bare Armv7-R, hardfloat
+[`thumbv8r-none-eabihf`](platform-support/armv8r-none-eabihf.md) | * | Thumb-mode Bare Armv8-R, hardfloat
 `i586-unknown-linux-gnu` | ✓ | 32-bit Linux (kernel 3.2+, glibc 2.17, original Pentium) [^x86_32-floats-x87]
 `i586-unknown-linux-musl` | ✓ | 32-bit Linux (musl 1.2.5, original Pentium) [^x86_32-floats-x87]
 [`i686-linux-android`](platform-support/android.md) | ✓ | 32-bit x86 Android ([Pentium 4 plus various extensions](https://developer.android.com/ndk/guides/abis.html#x86)) [^x86_32-floats-return-ABI]
@@ -419,8 +424,6 @@ target | std | host | notes
 [`thumbv5te-none-eabi`](platform-support/armv5te-none-eabi.md) | * |  | Thumb-mode Bare Armv5TE
 [`thumbv6-none-eabi`](platform-support/armv6-none-eabi.md) | * | | Thumb-mode Bare Armv6
 [`thumbv6m-nuttx-eabi`](platform-support/nuttx.md) | ✓ |  | ARMv6M with NuttX
-[`thumbv7a-none-eabi`](platform-support/armv7a-none-eabi.md) | * |  | Thumb-mode Bare Armv7-A
-[`thumbv7a-none-eabihf`](platform-support/armv7a-none-eabi.md) | * |  | Thumb-mode Bare Armv7-A, hardfloat
 [`thumbv7a-nuttx-eabi`](platform-support/nuttx.md) | ✓ |  | ARMv7-A with NuttX
 [`thumbv7a-nuttx-eabihf`](platform-support/nuttx.md) | ✓ |  | ARMv7-A with NuttX, hardfloat
 `thumbv7a-pc-windows-msvc` |  |  |
@@ -429,12 +432,9 @@ target | std | host | notes
 [`thumbv7em-nuttx-eabihf`](platform-support/nuttx.md) | ✓ |  | ARMv7EM with NuttX, hardfloat
 [`thumbv7m-nuttx-eabi`](platform-support/nuttx.md) | ✓ |  | ARMv7M with NuttX
 `thumbv7neon-unknown-linux-musleabihf` | ? |  | Thumb2-mode Armv7-A Linux with NEON, musl 1.2.5
-[`thumbv7r-none-eabi`](platform-support/armv7r-none-eabi.md) | * |  | Thumb-mode Bare Armv7-R
-[`thumbv7r-none-eabihf`](platform-support/armv7r-none-eabi.md) | * |  | Thumb-mode Bare Armv7-R, hardfloat
 [`thumbv8m.base-nuttx-eabi`](platform-support/nuttx.md) | ✓ |  | ARMv8M Baseline with NuttX
 [`thumbv8m.main-nuttx-eabi`](platform-support/nuttx.md) | ✓ |  | ARMv8M Mainline with NuttX
 [`thumbv8m.main-nuttx-eabihf`](platform-support/nuttx.md) | ✓ |  | ARMv8M Mainline with NuttX, hardfloat
-[`thumbv8r-none-eabihf`](platform-support/armv8r-none-eabihf.md) | * |  | Thumb-mode Bare Armv8-R, hardfloat
 [`wasm64-unknown-unknown`](platform-support/wasm64-unknown-unknown.md) | ? |  | WebAssembly
 [`wasm32-wali-linux-musl`](platform-support/wasm32-wali-linux.md) | ? |  | WebAssembly with [WALI](https://github.com/arjunr2/WALI)
 [`wasm32-wasip3`](platform-support/wasm32-wasip3.md) | ✓ |  | WebAssembly with WASIp3

--- a/src/doc/rustc/src/platform-support/arm-none-eabi.md
+++ b/src/doc/rustc/src/platform-support/arm-none-eabi.md
@@ -13,10 +13,13 @@ their own document.
 
 - Arm A-Profile Architectures
   - [`armv7a-none-eabi`](armv7a-none-eabi.md)
+  - [`thumbv7a-none-eabi` and `thumbv7a-none-eabihf`](armv7a-none-eabi.md)
 - Arm R-Profile Architectures
   - [`armv7r-none-eabi` and `armv7r-none-eabihf`](armv7r-none-eabi.md)
   - [`armebv7r-none-eabi` and `armebv7r-none-eabihf`](armebv7r-none-eabi.md)
   - [`armv8r-none-eabihf`](armv8r-none-eabihf.md)
+  - [`thumbv7r-none-eabi` and `thumbv7r-none-eabihf`](armv7r-none-eabi.md)
+  - [`thumbv8r-none-eabihf`](armv8r-none-eabihf.md)
 - Arm M-Profile Architectures
   - [`thumbv6m-none-eabi`](thumbv6m-none-eabi.md)
   - [`thumbv7m-none-eabi`](thumbv7m-none-eabi.md)

--- a/src/doc/rustc/src/platform-support/armv7a-none-eabi.md
+++ b/src/doc/rustc/src/platform-support/armv7a-none-eabi.md
@@ -1,7 +1,6 @@
 # `armv7a-none-eabi*` and `thumbv7a-none-eabi*`
 
-* **Tier: 2** (`armv7a-none-eabi` and `armv7a-none-eabihf`)
-* **Tier: 3** (`thumbv7a-none-eabi` and `thumbv7a-none-eabihf`)
+* **Tier: 2**
 * **Library Support:** core and alloc (bare-metal, `#![no_std]`)
 
 Bare-metal target for CPUs in the Armv7-A architecture family, supporting dual

--- a/src/doc/rustc/src/platform-support/armv7r-none-eabi.md
+++ b/src/doc/rustc/src/platform-support/armv7r-none-eabi.md
@@ -1,7 +1,6 @@
 # `armv7r-none-eabi*` and `thumbv7r-none-eabi*`
 
-* **Tier: 2** (`armv7r-none-eabi` and `armv7r-none-eabihf`)
-* **Tier: 3** (`thumbv7r-none-eabi` and `thumbv7r-none-eabihf`)
+* **Tier: 2**
 * **Library Support:** core and alloc (bare-metal, `#![no_std]`)
 
 Bare-metal target for CPUs in the Armv7-R architecture family, supporting dual

--- a/src/doc/rustc/src/platform-support/armv8r-none-eabihf.md
+++ b/src/doc/rustc/src/platform-support/armv8r-none-eabihf.md
@@ -1,7 +1,6 @@
 # `armv8r-none-eabihf` and `thumbv8r-none-eabihf`
 
-* **Tier: 2**: `armv8r-none-eabihf`
-* **Tier: 3**: `thumbv8r-none-eabihf`
+* **Tier: 2**
 * **Library Support:** core and alloc (bare-metal, `#![no_std]`)
 
 Bare-metal target for CPUs in the Armv8-R architecture family, supporting dual

